skiko/src/jvmMain/kotlin/org/jetbrains/skiko/SkiaLayer.jvm.kt

```diff
@@ -150,59 +150,59 @@ actual open class SkiaLayer internal constructor(
+            override fun mousePressed(e: MouseEvent) {
+                println("press")
+                skikoView?.onPointerEvent(toSkikoEvent(e))
+            }
+            override fun mouseReleased(e: MouseEvent) {
+                skikoView?.onPointerEvent(toSkikoEvent(e))
+            }
+            override fun mouseEntered(e: MouseEvent) {
+                skikoView?.onPointerEvent(toSkikoEvent(e))
             }
-            override fun mouseReleased(e: MouseEvent?) {
-                e!!
-                skikoView?.onPointerEvent(
-                    SkikoPointerEvent(e.x.toDouble(), e.y.toDouble(),
-                        if (e.button == 1) SkikoMouseButtons.LEFT else SkikoMouseButtons.RIGHT,
-                        SkikoPointerEventKind.UP,
-                        e
-                    )
-                )
+            override fun mouseExited(e: MouseEvent) {
+                skikoView?.onPointerEvent(toSkikoEvent(e))
             }
         })
-        backedLayer.addMouseMotionListener(object : MouseMotionAdapter() {
-            override fun mouseMoved(e: MouseEvent?) {
-                e!!
-                skikoView?.onPointerEvent(
-                    SkikoPointerEvent(e.x.toDouble(), e.y.toDouble(),
-                        SkikoMouseButtons.NONE,
-                        SkikoPointerEventKind.MOVE,
-                        e
-                    )
-                )
+
+        addMouseMotionListener(object : MouseMotionAdapter() {
+            override fun mouseDragged(e: MouseEvent) {
+                skikoView?.onPointerEvent(toSkikoEvent(e))
+            }
+            override fun mouseMoved(e: MouseEvent) {
+                skikoView?.onPointerEvent(toSkikoEvent(e))
             }
         })
-        backedLayer.addKeyListener(object : KeyAdapter() {
-            override fun keyPressed(e: KeyEvent?) {
-                e!!
-                skikoView?.onKeyboardEvent(
-                    SkikoKeyboardEvent(e.keyCode,
-                        SkikoKeyboardEventKind.DOWN,
-                        e
-                    )
-                )
+
+        addMouseWheelListener(object : MouseWheelListener {
+            override fun mouseWheelMoved(e: MouseWheelEvent) {
+                skikoView?.onPointerEvent(toSkikoEvent(e))
+            }
+        })
+
+        addKeyListener(object : KeyAdapter() {
+            override fun keyPressed(e: KeyEvent) {
+                skikoView?.onKeyboardEvent(toSkikoEvent(e))
+            }
+            override fun keyReleased(e: KeyEvent) {
+                skikoView?.onKeyboardEvent(toSkikoEvent(e))
+            }
+            override fun keyTyped(e: KeyEvent) {
+                skikoView?.onKeyboardEvent(toSkikoEvent(e))
+            }
+        })
+
+        addInputMethodListener(object : InputMethodListener {
+            override fun caretPositionChanged(e: InputMethodEvent) {
+                skikoView?.onInputEvent(toSkikoEvent(e))
```

Comment:
We probably shouldn't send input event on caret position change, as the text isn' changed.